piDIOComm.c

```diff
@@ -207,7 +209,7 @@ INT32U piDIOComm_sendCyclicTelegram(u8 devnum)
 	rcv_len = 3 * sizeof(u16) + i8uNumCounter[addr] * sizeof(u32);
 
 	ret = pibridge_req_io(addr, cmd, snd_buf, snd_len, in_buf, rcv_len);
-	if (ret) {
+	if (ret < 0) {
```

Comment:
Again, this needs to be `ret != rcv_len` so that too-short packets are treated as errors.